src/main/java/org/gridsuite/securityanalysis/server/dto/SubjectLimitViolationResultDTO.java

```diff
@@ -36,4 +41,19 @@ public static SubjectLimitViolationResultDTO toDto(SubjectLimitViolationEntity s
             .contingencies(contingencies)
             .build();
     }
+
+    // each SubjectLimitViolationResultDTO will return multiple line (one for each contingency)
+    public List<List<String>> toCsvRows(Map<String, String> translations) {
+        List<List<String>> csvRows = new ArrayList<>();
+        this.getContingencies().forEach(contingency -> {
+            List<String> csvRow = new ArrayList<>();
+            csvRow.add(this.getSubjectId());
+            csvRow.add(contingency.getContingency().getContingencyId());
+            csvRow.add(CsvExportUtils.translate(contingency.getContingency().getStatus(), translations));
+
+            csvRows.add(Stream.concat(csvRow.stream(), contingency.getLimitViolation().toCsvRow(translations).stream()).toList());
```

Comment:
wouldn't it be more efficient to do
```java
            csvRow.addAll(contingency.getLimitViolation().toCsvRow(translations));
            csvRows.add(csvRow);
```
?

Reply:
fixed